x-pack/plugins/ml/public/application/components/data_grid/data_grid.tsx

```diff
@@ -310,62 +322,72 @@ export const DataGrid: FC<Props> = memo(
-                            defaultMessage: 'Histogram charts',
-                          })}
-                        </EuiButtonEmpty>
-                      </EuiToolTip>
-                    ),
-                  }
-                : {}),
-            }}
-            popoverContents={popOverContent}
-            pagination={{
-              ...pagination,
-              pageSizeOptions: [5, 10, 25],
-              onChangeItemsPerPage,
-              onChangePage,
-            }}
-          />
-        </div>
+        <EuiMutationObserver
+          observerOptions={{ subtree: true, attributes: true, childList: true }}
+          onMutation={onMutation}
+        >
+          {(mutationRef) => (
+            <div className="mlDataGrid" ref={mutationRef}>
+              <EuiDataGrid
+                aria-label={isWithHeader(props) ? props.title : ''}
+                columns={columnsWithChartsActionized.map((c) => {
+                  c.initialWidth = 165;
+                  return c;
+                })}
+                columnVisibility={{ visibleColumns, setVisibleColumns }}
+                gridStyle={euiDataGridStyle}
+                rowCount={rowCount}
+                renderCellValue={renderCellValue}
+                sorting={{ columns: sortingColumns, onSort }}
+                toolbarVisibility={{
+                  ...euiDataGridToolbarSettings,
+                  ...(chartsButtonVisible
+                    ? {
+                        additionalControls: (
+                          <EuiToolTip
+                            content={i18n.translate(
+                              'xpack.ml.dataGrid.histogramButtonToolTipContent',
+                              {
+                                defaultMessage:
+                                  'Queries run to fetch histogram chart data will use a sample size per shard of {samplerShardSize} documents.',
+                                values: {
+                                  samplerShardSize: DEFAULT_SAMPLER_SHARD_SIZE,
+                                },
+                              }
+                            )}
+                          >
+                            <EuiButtonEmpty
+                              aria-pressed={chartsVisible === true}
+                              className={`euiDataGrid__controlBtn${
+                                chartsVisible === true ? ' euiDataGrid__controlBtn--active' : ''
+                              }`}
+                              data-test-subj={`${dataTestSubj}HistogramButton`}
+                              size="xs"
+                              iconType="visBarVertical"
+                              color="text"
+                              onClick={toggleChartVisibility}
+                              disabled={chartsVisible === undefined}
+                            >
+                              {i18n.translate('xpack.ml.dataGrid.histogramButtonText', {
```

Comment:
I recall `FormattedMessage` is a preferred way in react components according to Kibana guideline

Reply:
Fixed in 081ae1b.